source/components/triggering-changes-with-actions.md

```diff
@@ -156,3 +190,191 @@ a function instead.
 Actions in components allow you to
 decouple an event happening from how it's handled, leading to modular,
 more reusable components.
+
+## Handling Action Completion
+
+Often actions perform asynchronous tasks, such as making an ajax request to a server.
+Since actions are functions that can be passed in by a parent component, they are able to return values when called.
+The most common scenario is for an action to return a promise so that the component can handle the action's completion.
+
+In our user `button-with-confirmation` component we want to leave the confirmation modal open until we know that the
+operation has completed successfully.
+This is accomplished by expecting a promise to be returned from `onConfirm`.
+Upon resolution of the promise, we set a property used to indicate the visibility of the confirmation modal.
+
+```app/components/button-with-confirmation.js
+export default Ember.Component.extend({
+
+  actions: {
+    launchConfirmDialog() {
+      this.set('confirmShown', true);
+    },
+
+    submitConfirm() {
+      //call onConfirm with the value of the input field as an argument
+      this.get('onConfirm')().then() => {
```

Comment:
Would also do `var promise = this.get('onConfirm')();` and then `promise.then(`.